### PR TITLE
alps: init at 2021-09-29

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

This PR adds the package for [alps](https://git.sr.ht/~migadu/alps), a simple webmail server.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/alps/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+buildGoModule rec {
+  pname = "alps";
+  version = "d4c35f3c3157bece8e50fd95f2ee1081be30d7ae";
+
+  src = fetchFromSourcehut {
+    owner = "~migadu";
+    repo = "alps";
+    rev = "v${version}";
+    sha256 = "0000000000000000000000000000000000000000000000000000";
```
Comment (SuperSandro2000) [inline pkgs/servers/alps/default.nix:11]: You are missing a top-level entry and you can run nix-build locally to get this.

Comment (g0rdias) [reply pkgs/servers/alps/default.nix:11]: I was just looking into docs to figure this out, thank you! I'll push the changes up after I test.

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -4455,6 +4455,16 @@
     githubId = 1621335;
     name = "Andrew Trachenko";
   };
+  gordias = {
```
Comment (NotGordias) [inline maintainers/maintainer-list.nix:4458]: If it's preferable to separate this out into its own PR, I'd be happy to do that.

File: pkgs/servers/alps/default.nix (hunk 1 above)
Comment (NotGordias) [reply pkgs/servers/alps/default.nix]: Done, thanks!

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (legendofmiracles) [reply maintainers/maintainer-list.nix:4458]: Another PR is not needed, but a different commit `maintainers: add gordias` would be appreciated

Comment (NotGordias) [reply maintainers/maintainer-list.nix:4458]: Will do, thanks!

File: pkgs/servers/alps/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{ lib, buildGoModule, fetchFromSourcehut }:
+
+buildGoModule rec {
+  pname = "alps";
+  version = "d4c35f3c3157bece8e50fd95f2ee1081be30d7ae";
```
Comment (legendofmiracles) [inline pkgs/servers/alps/default.nix:5]: Version should be 2021-09-29, and the revision moved to the fetcher. Same with the PR title and commit please

Comment (NotGordias) [reply pkgs/servers/alps/default.nix]: Thank you! I've pushed the changes
